src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/DisplayFeatureLayers/DisplayFeatureLayers.xaml.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using ArcGISRuntime.Samples.Managers;
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Portal;
+using Esri.ArcGISRuntime.Security;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace ArcGISRuntime.WPF.Samples.DisplayFeatureLayers
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        name: "Display feature layers",
+        category: "Data",
+        description: "Display feature layers from various data sources.",
+        instructions: "Click the button on the toolbar to add feature layers, from different sources, to the map. Pan and zoom the map to view the feature layers.",
+        tags: new[] { "feature", "geodatabase", "geopackage", "layers", "service", "shapefile", "table" })]
+    [ArcGISRuntime.Samples.Shared.Attributes.OfflineData("1759fd3e8a324358a0c58d9a687a8578", "2b0f9e17105847809dfeb04e3cad69e0", "68ec42517cdd439e81b036210483e8e7", "15a7cbd3af1e47cfa5d2c6b93dc44fc2")]
+    public partial class DisplayFeatureLayers
+    {
+        public enum FeatureLayerSource
+        {
+            ServiceFeatureTable,
+            PortalItem,
+            Geodatabase,
+            Geopackaging,
+            Shapefile
+        }
+
+        public DisplayFeatureLayers()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Create a new map.
+            MyMapView.Map = new Map(BasemapStyle.ArcGISTopographic);
+
+            // Configure the feature layer selection box.
+            FeatureLayerCombo.ItemsSource = Enum.GetValues(typeof(FeatureLayerSource));
+            FeatureLayerCombo.SelectedItem = FeatureLayerSource.ServiceFeatureTable;
+        }
+
+        private async void FeatureLayerCombo_Selected(object sender, RoutedEventArgs e)
+        {
+            // Clear the existing FeatureLayer when a new FeatureLayer is selected.
+            MyMapView.Map.OperationalLayers.Clear();
+
+            switch (FeatureLayerCombo.SelectedItem)
+            {
+                case FeatureLayerSource.ServiceFeatureTable:
+                    await SetServiceFeatureTableFeatureLayer();
```

Comment:
We await in an async void without using a try/catch here. This should be addressed when removing the async void pattern.